raster/src/main/scala/geotrellis/raster/ResampleTarget.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import geotrellis.vector._
+import geotrellis.vector.io.json._
+import geotrellis.raster.reproject.Reproject
+
+import _root_.io.circe._
+import _root_.io.circe.syntax._
+
+import spire.math.Integral
+import spire.implicits._
+
+/** Represents a strategy/target for resampling */
+sealed trait ResampleTarget[N] {
```

Comment:
My main comment was that it sucks that using this forces all the `.resample` methods to become `.resample[N]` methods. I've realized that is "easily" fixed by moving the `N` type parameter from the trait definition to `apply` on the trait. So you never have to specify the span of target during definition but may encounter run-time errors if you try to resample a `Tile` to something outside of `Int` range (which is just as well basically).

Will report more as I play with this idea.